intro-HPC/ch_scoop.tex

```diff
@@ -0,0 +1,90 @@
+\chapter{SCOOP}
+\label{ch:scoop}
+
+SCOOP (Scalable COncurrent Operations in Python) is a distributed task module allowing
+concurrent parallel programming on various environments, from heterogeneous grids to supercomputers.
+
+It can used for projects that require lots of (small) tasks to be executed.
+
+The \lstinline|myscoop| script makes it very easy to use SCOOP, even in a multi-node setup.
+
+\section{Loading the module}
+
+Before using \lstinline|myscoop|, you first need to load the \lstinline|vsc-mympirun-scoop| module. We don't specify
+a version here (this is an exception, for most other modules you should, see \autoref{subsec:explicit-version-numbers})
+because newer versions might include important bug fixes or performance improvements.
+
+\begin{prompt}
+%\shellcmd{module load vsc-mympirun-scoop}%
+\end{prompt}
+
+\section{Write a worker script}
+
+A Python worker script implements both the main program, and (typically) the small task
+function that needs to be executed a large amount of times.
+
+Using the functionality provided by the scoop module, for example the \lstinline|futures.map|
+function (see also \url{https://scoop.readthedocs.org/}).
+
+First, the necessary imports need to be specified:
+
+\begin{code}{Python}
+import sys
+from scoop import futures
+\end{code}
+
+A Python function must be implemented for the core task, for example to compute the square of a number:
+
+\begin{code}{Python}
+def square(x):
+    return x*x
+\end{code}
+
+The main function then applies this simple function to a range of values specified as an argument.
+Note that it should be guarded by a conditional (\lstinline|if __name__ == "=__main__"|) to make sure it is only executed
+when executing the script (and not when importing from it).
```

Comment:
@redfast00 This example is incomplete, since we don't show how to define the main part... Can you try and fix that?